### PR TITLE
Add possibility to revoke specific session on the server side

### PR DESCRIPTION
**Pull request by tomnocon**
#123

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -67,16 +71,33 @@ module.exports = {
...
+        return new internals.RequestYar(request, settings, cache);
     };
 };
 
+internals.decorateServer = function (cache) {
+
+    return new internals.ServerYar(cache);
+};
+
+internals.ServerYar = class {
```
Comment (SimonSchick) [inline lib/index.js]: Can you change those to be `class ServerYar` and `RequestYar` respectively?
Anonymous classes can be annoying to debug :)

File: lib/index.js [hunk 2]
```diff
@@ -67,16 +71,33 @@ module.exports = {
...
+internals.decorateRequest = function (settings, cache) {
 
     return function (request) {
 
-        return new internals.Yar(request, settings, cache);
+        return new internals.RequestYar(request, settings, cache);
     };
 };
 
+internals.decorateServer = function (cache) {
```
Comment (SimonSchick) [inline lib/index.js]: This fn seems like a redundant layer of indirection for a 1 liner.

File: test/index.js [hunk 3]
```diff
@@ -1486,4 +1486,51 @@ describe('yar', () => {
...
+
+        // revoke session
+        await server.yar.revoke(res.result.sessionId);
+
+        const res2 = await server.inject({ method: 'GET', url: '/increment', headers: { cookie: cookie[1] } });
+        const res3 = await server.inject({ method: 'GET', url: '/increment', headers: { cookie: cookie[1] } });
+        expect(res2.result.value).to.equal(1);
+        expect(res3.result.value).to.equal(2);
+
+        return true;
```
Comment (SimonSchick) [inline test/index.js]: Correct me if I'm wrong but I think `return true` is redundant in those tests?

**Comment by tomnocon** [comment]
The build fails in unit tests because a new nodejs version (11) is used in the Travis node. This is not connected to the changes that I've made.

**Pull request merged**